### PR TITLE
Shooter intake states pid

### PR DESCRIPTION
**Pull request by LuisFerArredondo**
## Describe your changes

- [x] Bug fix
- [x] New feature

Explain what you did:
> Formatting work, added comments, giving a better name to stuff so is more readable and easier to understand

## Issue ticket (add number)

> https://github.com/FRC2706/2024-2706-Robot-Code/issues/...

## Checklist before requesting a review
- [x] I have performed a self-review of my code.
- [x] I have tested thoroughly.
- [x] I have written documentation and explained about my changes to the best of my ability.

### CODE REVIEW

File: src/main/java/frc/robot/commands/Shooter_PID_Tuner.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+import edu.wpi.first.wpilibj2.command.Command;
+import frc.robot.Config;
+import frc.robot.subsystems.ShooterSubsystem;
+
+public class Shooter_PID_Tuner extends Command {
+
+  //variable for setpoint
+  private final ShooterSubsystem shooter = ShooterSubsystem.getInstance() ;
+  private DoubleSupplier setPointInRPM;
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/commands/Shooter_PID_Tuner.java:17]: Can we just use double here?

**Review by weizhang-2706 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/Shooter_Voltage.java [hunk 2]
```diff
@@ -7,20 +7,19 @@
...
+import frc.robot.subsystems.ShooterSubsystem;
 
-public class Shooter_tuner extends Command {
+public class Shooter_Voltage extends Command {
 
   //variable for setpoint
-  private final Shooter shooter = Shooter.getInstance() ;
-  private DoubleSupplier setPoint;
+  private final ShooterSubsystem shooter = ShooterSubsystem.getInstance() ;
+  private DoubleSupplier setVoltage;
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/commands/Shooter_Voltage.java:17]: same comment here. Use double

**Review by weizhang-2706 — CHANGES_REQUESTED** [review]

File: src/main/java/frc/robot/commands/Shooter_Voltage.java [hunk 3]
```diff
@@ -32,18 +31,15 @@ public void initialize() {}
...
+    shooter.setVoltage(setVoltage.getAsDouble());
 
   }
   
 
   // Called once the command ends or is interrupted.
   @Override
   public void end(boolean interrupted) {
-
     shooter.setVoltage(0);
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/commands/Shooter_Voltage.java]: use stopMotor() of sparkmax  here
Add this method in the subsystem

File: src/main/java/frc/robot/commands/Shooter_PID_Tuner.java [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+  // Called every time the scheduler runs while the command is scheduled.
+  @Override
+  public void execute() {
+    shooter.setRPM(setPointInRPM.getAsDouble());
+  } 
+
+  // Called once the command ends or is interrupted.
+  @Override
+  public void end(boolean interrupted) {
+    shooter.setVoltage(0);
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/commands/Shooter_PID_Tuner.java]: use stopMotor()

File: src/main/java/frc/robot/robotcontainers/NewRobotContainer.java [hunk 5]
```diff
@@ -88,7 +93,8 @@ public NewRobotContainer() {
         )
     );
 
-    intake.setDefaultCommand(intake.autoIntake());
+    intake.setDefaultCommand(intake.defaultIntakeCommand());
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/robotcontainers/NewRobotContainer.java:78]: Add a boolean flag to switch between 1) simple manual intake/shooter 2) state based intake/shooter.

**Review by weizhang-2706 — CHANGES_REQUESTED** [review]

**Review by ErikCald — APPROVED** [review]

File: src/main/java/frc/robot/subsystems/ShooterSubsystem.java [hunk 6]
```diff
@@ -0,0 +1,184 @@
...
+    public void setRPM(double setPoint) {
+        m_pidController.setReference(setPoint, ControlType.kVelocity);
+    }
+
+    public void setVoltage(double setVolt) {
+        m_motor.setVoltage(setVolt);
+    }
+
+    public void stop(){
+        setVoltage(0);
```
Comment (weizhang-2706) [inline src/main/java/frc/robot/subsystems/ShooterSubsystem.java:100]: use m_motor.stopMotor() directly

**Review by weizhang-2706 — CHANGES_REQUESTED** [review]

**Review by weizhang-2706 — APPROVED** [review]

**Pull request merged**